src/graph/expression_graph.h

```diff
@@ -112,97 +123,167 @@ class Tensors {
 
 typedef std::map<Type, Ptr<Parameters>> ElementTypeParamsMap; // keep it sorted, hence map not unordered map
 
+/**
+ * @brief Main implementation of a computation graph
+ *
+ *  Keeps a record of data (tensors) & all operations
+ *  Each operation in a computation graph is a Node.
+ *  Each Node defines its forward and backward steps.
+ */
 class ExpressionGraph : public std::enable_shared_from_this<ExpressionGraph> {
-  size_t count_{0};
+  size_t count_{0};  /**< @brief counter for nodes in the graph; hold current node index */
```

Comment:
I think we don't need doxygen-style comments for private attributes as they will not be displayed in a generated documentation anyway, so maybe just simply use `// `.

I think we can skip `@brief` in inline comments in general, what do you think?

I would recommend using single convention for in-line doxygen docs, so `///< comment` instead of `/**< comment */`.

Reply:
I agree. We can skip private attributes and `@brief` in inline comments in general.